src/lazyoperations.jl

```diff
@@ -115,7 +115,12 @@ _compatible_sizes((A, B)) = (size(A, 2) == size(B, 1))
 
 # Implements the mixed-product property for the Kronecker product and matrix
 # multiplication
-function copy(M::Mul{ApplyLayout{typeof(kron)},ApplyLayout{typeof(kron)},<:AbstractMatrix,<:AbstractMatrix})
+function copy(M::Mul{
```

Comment:
Why did you change the type signature? There is no reason to restrict to `Kron` as the `ApplyLayout` takes care of that/

Reply:
i need to dispatch on the fact that both arguments here are `Kron` types or else it'll just go straight to the shuffle algo

Reply:
No the first two `ApplyLayout{typeof(kron)}` tell you both are lazy kronecker products, you don't need to worry about the types of matrices.

Reply:
oh you're right 

I think I got carried away when I fixed the vec-trick dispatch and just applied the same strategy here